### ISSUE TITLE
--s3-profile does not work when AWS_PROFILE does

### ISSUE THREAD

**Issue by nh2** [rclone/rclone#4757]
#### What is the problem you are having with rclone?

I believe there's a bug somewhere in between `--s3-profile` and `AWS_PROFILE`. The latter works, the former doesn't.

I'm using AWS s3 with MFA and a corresponding profile declared in `~/.aws/credentials`.

`~/.rclone.conf`:

```
[mys3]
type = s3
provider = AWS
env_auth = true
region = us-east-1
acl = private
```

`~/.aws/credentials`:

```ini
[my-aws-profile-mfa]
aws_access_key_id = BBBBBBBBBBBBBBBBBBBB
aws_secret_access_key = bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
aws_session_token = sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss
```

This works:

```sh
AWS_PROFILE=my-aws-profile-mfa rclone lsd 'mys3:'
```

This fails:

```sh
rclone --s3-profile my-aws-profile-mfa lsd 'mys3:'
```

with `AccessDenied`:

```
2020/11/10 15:49:49 ERROR : : error listing: AccessDenied: Access Denied
  status code: 403, request id: AAAAAAAAAAAAAAAA, host id: BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB
2020/11/10 15:49:49 Failed to lsd with 2 errors: last error was: AccessDenied: Access Denied
  status code: 403, request id: AAAAAAAAAAAAAAAA, host id: BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB
```

I used `strace` to determine that either invocation reads `~/.aws/credentials` 4 times, so the problem must be something else.

---

The docs of `--s3-profile` say:

> If empty it will default to the environment variable `"AWS_PROFILE"` or `"default"` if that environment variable is also not set.

however this doesn't seem to work as described.

#### What is your rclone version (output from `rclone version`)

```
rclone 1.53.0
- os/arch: linux/amd64
- go version: go1.15.2
```

on NixOS 20.09.

**Comment by dimonx3** [comment]
I have 2 named profile in ~/.aws/credentials.
I try specify --s3-profile as command parameter but it does not work.
`clone size test:/foo/bar --s3-profile 'my_profile' -vvv`
Failed to size: NoCredentialProviders: no valid providers in chain. Deprecated.

**Comment by ncw** [comment]
Can you give this a go please @dimonx3 

I think it will fix the problem

[v1.54.0-beta.5073.408db6778.fix-4757-s3-profile](https://beta.rclone.org/branch/fix-4757-s3-profile/v1.54.0-beta.5073.408db6778.fix-4757-s3-profile/) on branch [fix-4757-s3-profile](https://github.com/rclone/rclone/tree/fix-4757-s3-profile) (uploaded in 15-30 mins)

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.55
